glean_parser/templates/go_server.jinja2

```diff
@@ -0,0 +1,218 @@
+{# The final Go code is autogenerated, but this template is not. Please file bugs! #}
+package glean
+
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// AUTOGENERATED BY glean_parser v{{ parser_version }}. DO NOT EDIT.
+
+// required imports
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// log type string used to identify logs to process in the Moz Data Pipeline
+var gleanEventMozlogType string = "glean-server-event"
+
+type GleanEventsLogger struct {
+	AppID             string // Application Id to identify application per Glean standards
+	AppDisplayVersion string // Version of application emitting the event
+	AppChannel        string // Channel to differentiate logs from prod/beta/staging/devel
+}
+
+// exported type for public method parameters
+type RequestInfo struct {
+	UserAgent string
+	IpAddress string
+}
+
+// default empty values will be omitted in json from ping struct definition
+var defaultRequestInfo = RequestInfo{
+	UserAgent: "",
+	IpAddress: "",
+}
+
+// structs to construct the glean ping
+type clientInfo struct {
+	TelemetrySDKBuild string `json:"telemetry_sdk_build"`
+	FirstRunDate      string `json:"first_run_date"`
+	OS                string `json:"os"`
+	OSVersion         string `json:"os_version"`
+	Architecture      string `json:"architecture"`
+	AppBuild          string `json:"app_build"`
+	AppDisplayVersion string `json:"app_display_version"`
+	AppChannel        string `json:"app_channel"`
+}
+
+type pingInfo struct {
+	Seq       int    `json:"seq"`
+	StartTime string `json:"start_time"`
+	EndTime   string `json:"end_time"`
+}
+
+type ping struct {
+	DocumentNamespace string `json:"document_namespace"`
+	DocumentType      string `json:"document_type"`
+	DocumentVersion   string `json:"document_version"`
+	DocumentID        string `json:"document_id"`
+	UserAgent         string `json:"user_agent,omitempty"`
+	IpAddress         string `json:"ip_address,omitempty"`
+	Payload           string `json:"payload"`
+}
+
+type metrics map[string]map[string]interface{}
+
+type pingPayload struct {
+	ClientInfo clientInfo   `json:"client_info"`
+	PingInfo   pingInfo     `json:"ping_info"`
+	Metrics    metrics      `json:"metrics"`
+	Events     []gleanEvent `json:"events"`
+}
+
+type gleanEvent struct {
+	Category  string                 `json:"category"`
+	Name      string                 `json:"name"`
+	Timestamp int64                  `json:"timestamp"`
+	Extra     map[string]interface{} `json:"extra"`
```

Comment:
event extras are a string->string map. the types (string, bool, quantity) are client-side only and get stringified before sending.